### PR TITLE
jax frontend histogram bin edges + test

### LINKED ISSUE #16983: histogram_bin_edges

### PR DESCRIPTION
**Pull request by mam15**
Close #16983

### CODE REVIEW

File: ivy/functional/backends/paddle/device.py
Comment (MahmoudAshraf97) [inline ivy/functional/backends/paddle/device.py:1]: Please remove this change and if needed for a bug fix, keep it in a separate PR

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/frontends/jax/numpy/statistical.py [hunk 1]
```diff
@@ -23,6 +23,30 @@ def einsum(
     return ivy.einsum(subscripts, *operands, out=out)
 
 
+@to_ivy_arrays_and_back
+def histogram_bin_edges(a, bins=10, range=None, weights=None):
+    del weights
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/jax/numpy/statistical.py:28]: May be this could be removed as having this variable doesn't effect the function

Comment (bipinKrishnan) [inline ivy/functional/frontends/jax/numpy/statistical.py:28]: Is there any reason for not supporting the `weights` argument?

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py [hunk 2]
```diff
@@ -16,6 +16,100 @@
...
+        )
+    )
+    density = draw(st.booleans())
+    return (
+        a,
+        bins,
+        range,
+        weights,
+        density,
+        dtype_input,
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py:109]: Could you specify the case for which this helper function is required. While going through the code I felt like we should be able to get these data from existing array helpers. Let me know if that's not the case :slightly_smiling_face:

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_statistical.py [hunk 3]
```diff
@@ -970,8 +1005,8 @@ def test_jax_nanmedian(
     test_flags,
 ):
     input_dtype, x, axis = dtype_x_axis
-    # TODO: overwrite as a boolean when there's a way around jax.numpy.nanquantile does not
-    #  support overwrite_input=True.
+    # TODO: overwrite as a boolean when there's a way around
+    # jax.numpy.nanquantile does not support overwrite_input=True.
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_statistical.py:1009]: Seems like these are not related to your changes, can you remove these modifications?

**Comment by bipinKrishnan** [comment]
The test for your function is failing due to missing kwarg, can fix this @mam15

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_statistical.py [hunk 4]
```diff
@@ -65,6 +66,40 @@ def test_jax_einsum(
...
+    fn_tree="jax.numpy.histogram_bin_edges",
+    values=histogram_helper(),
+)
+def test_jax_histogram_bin_edges(
+    *,
+    values,
+    on_device,
+    fn_tree,
+    frontend,
+    test_flags,
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_statistical.py:80]: I think you missed an argument here, `backend_fw` I guess. Just refer to other test implementations and your test failure outputs in the CI

File: ivy/functional/frontends/jax/numpy/statistical.py (hunk 1 above)
Comment (bipinKrishnan) [reply ivy/functional/frontends/jax/numpy/statistical.py:28]: > Is there any reason for not supporting the `weights` argument?

Hey can you address this question and the one below it as well :slightly_smiling_face:

**Comment by bipinKrishnan** [comment]
I've left some comments in the review, clarifying those would allow merging the PR quicker. Thanks :slightly_smiling_face:

**Comment by mam15** [comment]
> I've left some comments in the review, clarifying those would allow merging the PR quicker. Thanks 🙂

@bipinKrishnan I'm so sorry! I had replied to those comments weeks ago but forgot to submit my review, so they were pending and you couldn't see them 🤦‍♂️

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_statistical.py [hunk 5]
```diff
@@ -65,6 +66,42 @@ def test_jax_einsum(
     )
 
 
+# histogram_bin_edges
+@handle_frontend_test(
+    fn_tree="jax.numpy.histogram_bin_edges",
+    values=histogram_helper(),
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_statistical.py:72]: Most of the things returned from the `histogram_helper` could be obtained using existing helper functions, for example `a`, `bins`, `weights` etc are normal arrays which could be easily obtained using existing helpers. I would recommend going through the [testing deep dive](https://unify.ai/docs/ivy/overview/deep_dive/ivy_tests.html) or the [hypothesis helpers](https://github.com/unifyai/ivy/tree/main/ivy_tests/test_ivy/helpers/hypothesis_helpers) to learn about the functions that suit your use-case.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
